File: 3_semestre/estrutura_de_dados/Aulas/aula08/src/decToBin.c
/*
    Programa que transforma decimal em binário
*/

#include <stdio.h>

#define TRUE 1
#define FALSE 0

int bin[32] = {};
int cont = 0;

int decToBin(int dec){

    while(TRUE){

        if((dec % 2) == 0){
            bin[cont] = 0;                                     
        } else {
            bin[cont] = 1;
        }
        
        dec = dec / 2;
        cont++;

        if(dec == 0){
            break;
        }

    }
}

void main(void){

    int dec = 10;

    decToBin(dec); 

    while(cont != 0){
        printf("%d\n", bin[cont--]);
    }
}

File: 3_semestre/estrutura_de_dados/Aulas/aula08/src/decToBin.py
bin = []

def decToBin(dec):
    
    while(True):
        
        if((dec % 2) == 0):
            bin.append(0)
        else:
            bin.append(1)

        dec = dec // 2

        if(dec == 0):
            break
            
decToBin(10)

while bin:
    print(str(bin.pop()), end= '')